### ISSUE TITLE
Generated inventory for Ansible always uses ansible_ssh_* vars

### ISSUE THREAD

**Issue by lpancescu** [mitchellh/vagrant#7067]
Vagrant generates an inventory file for Ansible, containing values for ansible_ssh_host and ansible_ssh_port. However, in Ansible 2.0 these variables have been renamed to ansible_host and ansible_port. Perhaps the inventory generation needs to be handled depending on the Ansible version installed on the system? This breaks automatic rebooting via Ansible (I'm using a wait_for local action, where I need to specify the correct host and port).

**Comment by gildegoma** [comment]
@lpancescu thank you for reporting this.

I confirm that his point will be addressed (like all other parameters that have been deprecated as of Ansible 2.0). Please see #6570 for more details.

Please note also that these legacy "ssh" parameter names have been _deprecated_ in Ansible 2.0, and probably will be fully removed in Ansible 2.2. For the moment, I think you should be able to workaround your local "wait_for" use case by sticking to these old "ssh" variable names. If this is not working for you, could you please provide a `Vagrantfile` and an Ansible playbook file that simply reproduce the problem, as suggested in the [new vagrant issue template](https://github.com/mitchellh/vagrant/blob/master/.github/ISSUE_TEMPLATE.md).

**Comment by lpancescu** [comment]
@gildegoma thank you for your fast answer (sorry for answering so late, I've been debugging why CentOS doesn't come up reliably with the VirtualBox provider for the last few days).

I just checked with a `debug: var` statement in Ansible, and you're right: the "new" and the "deprecated" values have the same values, even if the inventory file set the deprecated ones. I see different behavior when using the deprecated variables in the argument list to `wait_for` (Ansible throws a Python exception) than when using the new ones (it just waits for the timeout to expire). I'm not sure what's wrong, I filed [issue 14675](https://github.com/ansible/ansible/issues/14675) with the Ansible project.
